plugins/fabric8-multi-tenant-manager/src/test/java/com/redhat/che/multitenant/tenantdata/TenantDataCacheLoaderTest.java

```diff
@@ -0,0 +1,158 @@
+  @Mock private HttpJsonRequest httpJsonRequest;
+
+  private TenantDataCacheLoader cacheLoader;
+  private TenantDataCacheLoader standaloneCacheLoader;
+
+  @BeforeMethod
+  public void setUp() throws Exception {
+    when(httpJsonRequestFactory.fromUrl(any(String.class))).thenReturn(httpJsonRequest);
+    when(httpJsonRequest.setMethod(any(String.class))).thenReturn(httpJsonRequest);
+    when(httpJsonRequest.setAuthorizationHeader(any(String.class))).thenReturn(httpJsonRequest);
+    when(httpJsonRequest.request()).thenReturn(httpJsonResponse);
+
+    cacheLoader = new TenantDataCacheLoader(httpJsonRequestFactory, ENDPOINT, NAMESPACE, false);
+    standaloneCacheLoader =
+        new TenantDataCacheLoader(httpJsonRequestFactory, ENDPOINT, NAMESPACE, true);
+  }
+
+  @Test
+  public void shouldParseJsonAndReturnTenantData() throws Exception {
+    // Given
+    String jsonResponse =
+        generateResponse(ROUTE_PREFIX, CLUSTER_URL, NAMESPACE, NAMESPACE_TYPE, false);
+    when(httpJsonResponse.asString()).thenReturn(jsonResponse);
+    TenantDataCacheKey cacheKey = new TenantDataCacheKey("token", "che");
+
+    // When
+    UserCheTenantData data = cacheLoader.load(cacheKey);
+
+    // Then
+    assertEquals(data.getRouteBaseSuffix(), ROUTE_PREFIX);
+    assertEquals(data.getClusterUrl(), CLUSTER_URL);
+    assertEquals(data.getNamespace(), NAMESPACE);
+    assertEquals(data.isClusterCapacityExhausted(), false);
+  }
+
+  @Test(expectedExceptions = InfrastructureException.class)
+  public void shouldThrowInfrastructureExceptionWhenFailToGetResponse() throws Exception {
+    // Given
+    when(httpJsonRequest.request()).thenThrow(new IOException());
+    TenantDataCacheKey cacheKey = new TenantDataCacheKey("token", "che");
+
+    // When
+    cacheLoader.load(cacheKey);
+  }
+
+  @Test(expectedExceptions = InfrastructureException.class)
+  public void shouldThrowInfrastructureExceptionWhenNoNamespaceMatchesCacheKey() throws Exception {
+    // Given
+    String jsonResponse =
+        generateResponse(ROUTE_PREFIX, CLUSTER_URL, NAMESPACE, NAMESPACE_TYPE, false);
+    when(httpJsonResponse.asString()).thenReturn(jsonResponse);
+    TenantDataCacheKey cacheKey = new TenantDataCacheKey("token", "not-che");
+
+    // When
+    cacheLoader.load(cacheKey);
+  }
+
+  @Test(expectedExceptions = InfrastructureException.class)
+  public void shouldThrowInfrastructureExceptionWhenResponseFailsToValidate() throws Exception {
+    // Given
+    String jsonResponse = generateBadResponse(ROUTE_PREFIX, NAMESPACE, NAMESPACE_TYPE, false);
+    when(httpJsonResponse.asString()).thenReturn(jsonResponse);
+    TenantDataCacheKey cacheKey = new TenantDataCacheKey("token", "che");
+
+    // When
+    cacheLoader.load(cacheKey);
+  }
+
+  @Test
+  public void standaloneShouldReturnDefaultDataWithCheNamespaceType() throws Exception {
+    // Given
+    TenantDataCacheKey cacheKey = new TenantDataCacheKey("token", "che");
```

Comment:
Consider adding test that checks case when json is valid but an attribute is null instead of the object. This potentially can emit NPE. For example 
 ```json
{
    "data": {
        "attributes": null
    }
}
```

Reply:
I've added a test cast to cover the NPE described above, that *should* cover this. Apart from that if any of the fields in namespace is null that should cause a `ValidationException`.

Reply:
Added a javadoc to the `getNamespaces()` method to mention the potential NPE.